OSDOCS-5243 Adding CLI Manager docs (#81448)
openshift/openshift-docs: modules/cli-manager-custom-index-krew.adoc

:_mod-docs-content-type: PROCEDURE
[id="cli-manager-custom-index_{context}"]
= Adding the {cli-manager} custom index to Krew

Use the terminal to add the CLI manager custom index to Krew. This is required for the {cli-manager} to function correctly.

[NOTE]
====
If you use self-signed certificates, mark the certificate as trusted on your local operating system to use Krew.
====

.Prerequisites

* link:https://krew.sigs.k8s.io/docs/user-guide/setup/install[Krew is installed].
* The CLI Manager is installed.
* You are logged in to {product-title} as a user with the `cluster-admin` role.

.Procedure

. To establish the `ROUTE` variable, enter the following command:

[source,terminal]
----
ROUTE=$(oc get route/openshift-cli-manager -n openshift-cli-manager-operator -o=jsonpath='{.spec.host}')
----

. Name the custom index by running the following command:

[source,terminal]
----
# <custom_index_name>=ocp
----

. To add the custom index to Krew, by entering the following command:

[source,terminal]
----
# oc krew index add $<custom_index_name> https://$ROUTE/cli-manager
----

. To update Krew, enter the following command:

[source,terminal]
----

> # oc krew update

Author: @ardaguclu @kasturinarra Would it cause any problems if a user already had Krew installed?

> ----

Reviewer: May be add a step which reads `make sure no errors are seen when oc kew update` is run.

Author: What kind of errors could there be and how would users troubleshoot them?

Reviewer 2: +1 it would be good to show an example too of what a successful output would be after running this command.

Reviewer: so a successful run would indicate output something as below, But let me get you a more concrete one may be tomorrow.

I1112 23:55:39.074785 236313 client.go:835] Running 'oc krew update'
  Updated the local copy of plugin index.
  Updated the local copy of plugin index "ocp".

If there is a new plugin detected, sometimes it says that new plugin ocp/oc-mirror detected..

// Module included in the following assemblies:
//
// * cli_reference/cli_manager/cli-manager-install.adoc

:_mod-docs-content-type: PROCEDURE
[id="cli-manager-custom-index_{context}"]
= Adding the {cli-manager} custom index to Krew

> You can use the terminal to add the CLI manager custom index to Krew. This is required for the {cli-manager} to function correctly and only needs to be done once. The custom index should be added to Krew to connect Krew to the {cli-manager} binaries and is what enables the {cli-manager} to work in disconnected environments.

Reviewer 3: ```suggestion
You can use the terminal to add the CLI manager custom index to Krew. This procedure is required for the {cli-manager} to function correctly and needs to be done only once. 

The custom index connects Krew to the {cli-manager} binaries and enables the {cli-manager} to work in disconnected environments.
```

:_mod-docs-content-type: PROCEDURE
[id="cli-manager-custom-index_{context}"]
= Adding the {cli-manager} custom index to Krew

You can use the terminal to add the CLI manager custom index to Krew. This procedure is required for the {cli-manager} to function correctly and needs to be done only once.

The custom index connects Krew to the {cli-manager} binaries and enables the {cli-manager} to work in disconnected environments.

[NOTE]
====
If you use self-signed certificates, mark the certificate as trusted on your local operating system to use Krew.
====

.Prerequisites

* link:https://krew.sigs.k8s.io/docs/user-guide/setup/install[Krew is installed].
* The {cli-manager} is installed.

.Procedure

. To establish the `ROUTE` variable, enter the following command:
+
[source,terminal]
----
$ ROUTE=$(oc get route/openshift-cli-manager -n openshift-cli-manager-operator -o=jsonpath='{.spec.host}')
----

. To add the custom index to Krew, enter the following command:
+
[source,terminal]
----
$ oc krew index add <custom_index_name> https://$ROUTE/cli-manager
----

. To update Krew, enter the following command and check for any errors:
+
[source,terminal]
----
$ oc krew update
----

> 

Reviewer 4: Probably align this output in a future PR.

// Module included in the following assemblies:
//
// * cli_reference/cli_manager/cli-manager-install.adoc

:_mod-docs-content-type: PROCEDURE
[id="cli-manager-custom-index_{context}"]
= Adding the CLI Manager custom index to Krew

> Use the terminal to add the CLI manager custom index to Krew.

Reviewer 2: We talked about this too, but should add some details on why you want (need?) to do this. Is this optional, or is this a required step?

.Prerequisites

* link:https://krew.sigs.k8s.io/docs/user-guide/setup/install[Krew is installed].
* The CLI Manager is installed.
* You are logged in to {product-title} as a user with the `cluster-admin` role.

> * You are using {product-title} 4.17 or later.

Reviewer 2: Same comment about not needing this line

.Procedure

> . Execute the following command:

Reviewer 2: Should give some detail about what you're doing here.

Also consider using our normal format of "To <x>, run the following command:" or "<x> by running the following command:" for each of your steps throughout this PR

https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command

> 

Reviewer 2: We talked about this already but there needs to be "+" between the step and the code block for these to be indented/numbered correctly

// Module included in the following assemblies:
//
// * cli_reference/cli_manager/cli-manager-install.adoc

:_mod-docs-content-type: PROCEDURE
[id="cli-manager-custom-index_{context}"]
= Adding the {cli-manager} custom index to Krew

Use the terminal to add the CLI manager custom index to Krew. This is required for the {cli-manager} to function correctly.

> NOTE

Reviewer 2: This should fix at least the existing build error
```suggestion
[NOTE]
```

====

.Prerequisites

* link:https://krew.sigs.k8s.io/docs/user-guide/setup/install[Krew is installed].
* The CLI Manager is installed.
* You are using {product-title} 4.17 or later.

.Procedure

. To establish the `ROUTE` variable, enter the following command:

[source,terminal]
----
ROUTE=$(oc get route/openshift-cli-manager -n openshift-cli-manager-operator -o=jsonpath='{.spec.host}')
----

. Name the custom index by running the following command:

[source,terminal]
----
# <custom_index_name>=ocp
----

. To add the custom index to Krew, by entering the following command:

[source,terminal]
----
# oc krew index add $<custom_index_name> https://$ROUTE/cli-manager
----

.. If you see certificate authority errors run the following command to get the certificate authority data and write it to your local machine.

[source,terminal]
----
# echo "$(oc config view --minify --flatten -o jsonpath='{.clusters[0].cluster.certificate-authority-data}' | base64 --decode)" | sudo tee /etc/pki/ca-trust/source/anchors/cli.crt > /dev/null
----

.. To update the certificate authority for it to be trusted, run the following command:

[source,terminal]
----

> # sudo update-ca-trust

Author: Are there workarounds for the operating systems we support? Or should the procedure work as intended for those?

> # sudo update-ca-trust

Reviewer 5: It would be difficult to provide solutions for all the operating systems which Krew supports.
